chalk-engine/src/logic.rs

```diff
@@ -504,20 +505,18 @@ impl<'forest, C: Context + 'forest, CO: ContextOps<C> + 'forest> SolveState<'for
                             self.on_subgoal_selected(strand)?;
                             continue;
                         }
-                        SubGoalSelection::NoRemainingSubgoals => {
-                            match self.on_no_remaining_subgoals(strand) {
+                        selection @ SubGoalSelection::NoRemainingSubgoals
+                        | selection @ SubGoalSelection::Floundered => {
```

Comment:
This can actually just be simplified to `NotSelected` and `Selected`. And we know if it floundered based on if there are any floundered subgoals.

Reply:
done